ethcore/src/engines/authority_round/mod.rs

```diff
@@ -806,19 +813,25 @@ impl AuthorityRound {
 			return;
 		}
 
-		if let (true, Some(me)) = (current_step > parent_step + 1, self.signer.read().as_ref().map(|s| s.address())) {
+		if let (true, me) = (current_step > parent_step + 1, self.address()) {
 			debug!(target: "engine", "Author {} built block with step gap. current step: {}, parent step: {}",
 				   header.author(), current_step, parent_step);
-			let mut reported = HashSet::new();
-			for step in parent_step + 1..current_step {
-				let skipped_primary = step_proposer(validators, header.parent_hash(), step);
-				// Do not report this signer.
-				if skipped_primary != me {
-					// Stop reporting once validators start repeating.
-					if !reported.insert(skipped_primary) { break; }
-					self.validators.report_benign(&skipped_primary, set_number, header.number());
- 				}
- 			}
+			if validators.contains(header.parent_hash(), &me) {
+				let mut reported = HashSet::new();
+				for step in parent_step + 1..current_step {
+					let skipped_primary = step_proposer(validators, header.parent_hash(), step);
+					// Do not report this signer.
+					if skipped_primary != me {
+						// Stop reporting once validators start repeating.
+						if !reported.insert(skipped_primary) { break; }
+						self.validators.report_benign(&skipped_primary, set_number, header.number());
+					} else {
+						trace!(target: "engine", "Primary that skipped is self, not self-reporting. Own address: {}", me);
+					}
+				}
+			} else {
+				warn!(target: "engine", "We are not part of the validator set so not reporting (skipped steps). Own address: {}", self.address());
```

Comment:
This definitely shouldn't be printed if `engine_signer` is not configured.

Reply:
My thinking here was: something called `report_skipped()` which I assumed shouldn't ever happen unless `engine_signer` is a-ok, so if they're set up to sign, but not part of the validator set then perhaps there's a bug worth knowing about?